pkg/cloudprovider/aws/fake/ec2api.go

```diff
@@ -44,19 +44,28 @@ type EC2Behavior struct {
 	CalledWithCreateLaunchTemplateInput set.Set
 	Instances                           sync.Map
 	LaunchTemplates                     sync.Map
+	ShouldTriggerInsufficientCapacity   bool
```

Comment:
Thoughts on collapsing this and InsufficientCapacityInstanceType into a single configurable concept
`InsufficientCapacityExceptionInstanceTypes sets.Set`